src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryFileValidator.cs

```diff
@@ -0,0 +1,172 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Data;
+using System.Data.Common;
+using System.Data.SqlClient;
+using System.Globalization;
+using System.IO;
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.SqlServer.Management.Sdk.Sfc;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.Connection.ReliableConnection;
+using Microsoft.SqlTools.ServiceLayer.FileBrowser;
+
+namespace Microsoft.SqlTools.ServiceLayer.DisasterRecovery
+{
+    /// <summary>
+    /// Validate selected file paths for backup/restore operations
+    /// </summary>
+    public static class DisasterRecoveryFileValidator
+    {
+        private const string localSqlServer = "(local)";
+        private const string localMachineName = ".";
+
+        public static bool ValidatePaths(FileBrowserValidateEventArgs args, out string errorMessage)
+        {
+            errorMessage = "";
+            bool result = true;
+            DbConnection dbConn = null;
+            ConnectionInfo connInfo;
+
+            if (args != null)
+            {
+                ConnectionService.Instance.TryFindConnection(args.OwnerUri, out connInfo);
+                SqlConnection conn = null;
```

Comment:
>SqlConnection conn = null; [](start = 16, length = 26)

I would define this variable on line #46 since it doesn't appear to be used outside that scope.  typically code is cleaner if variables are not defined until they are needed and only scoped to the block they are needed.